ssl/statem/statem_dtls.c

```diff
@@ -926,6 +927,19 @@ WORK_STATE dtls_wait_for_dry(SSL *s)
     }
 
     if (ret == 0) {
+        /*
+         * We're not expecting any more messages from the peer at this point -
+         * but we could get an alert. If an alert is waiting then we will never
+         * return successfully. Therefore we attempt to read a message. This
+         * should never succeed but will process any waiting alerts.
+         */
```

Comment:
This is at least a little confusing. As far as I understand, ret == 0 means we're not dry yet, that we still have messages in the send queue that the other side did not acknowledge. I'm not sure why it says we're not expecting any message.

Reply:
We wait for a dry before sending a CCS. At this point the peer shouldn't be sending us anything in a normal handshake because it is our "turn" to talk. But if something went wrong then they could have sent us an alert. Therefore we attempt to read a message. We should never get one if the peer is behaving well because it should be waiting for us - but it will at least correctly process an alert that we have been sent.

Reply:
This looks like a general function to wait for a dry event. If this is only used just before sending a CCS, this should probably be documented.

Reply:
Actually this can also happen before a ServerDone is sent, but the same argument applies (i.e. we are at a point in the handshake where the peer shouldn't be sending us anything). I've added a comment explaining this in front of the function.